i6_models/samplers/log_uniform.py

```diff
@@ -0,0 +1,30 @@
+__all__ = ["LogUniformSampler"]
+
+
+import math
+
+import torch
+from torch import nn
+
+
+class LogUniformSampler(nn.Module):
+    def __init__(self, num_classes):
+        super().__init__()
+
+        # assumes count-sorted vocabulary, descending
+        self.num_classes = num_classes
+
+        # approximately zipf distribution
+        self._distribution = [
+            (math.log1p(w + 1) - math.log1p(w)) / math.log1p(self.num_classes) for w in range(self.num_classes)
+        ]
+        self._distribution = torch.tensor(self._distribution).clamp(min=1e-10)
+        self._distribution /= self._distribution.sum()
```

Comment:
Why first make a Python list of Python floats? Why not use Torch right away?
```suggestion
        ws = torch.arange(self.num_classes, dtype=torch.get_default_dtype())
        self._distribution = (torch.log1p(ws + 1) - torch.log1p(ws)) / torch.log1p(torch.tensor(self.num_classes))
        self._distribution.clamp_(min=1e-10)
        self._distribution /= self._distribution.sum()
```